### PR TITLE
Break up and flatten internal package.

### PR DESCRIPTION
**Pull request by matthewdale**
## Summary
Break up the `internal` package into separate packages per use and flatten the packages in the internal directory into one level. The only expected behavior change is that `assert.Soon` (renamed from `helpers.AssertSoon`) calls `t.Error` instead of `t.Fail`. Otherwise, it's all moving code and renaming packages.

Separate the code in the existing `internal` package into the following packages:
* `internal/bsonutil`
* `internal/csfle`
* `internal/csot`
* `internal/errutil`
* `internal/handshake`
* `internal/httputil`

Flatten and rename packages:
* `internal/testutil/monitor` -> `internal/eventtest`
* `internal/randutil/rand` -> `internal/rand`
* `internal/testutil/israce` -> `internal/israce`
* `internal/testutil/helpers` -> `internal/spectest` (`helpers.AssertSoon` moves to `assert.Soon`)
* `internal/testutil` -> `internal/integtest`

Additional changes:
* Move some code from `internal/` that's only used in a single package into the package where it's used.
* Remove unused `internal.MultiError` function.

## Background & Motivation
The `internal` directory shouldn't really be a package, just a directory that prevents external packages from importing the packages it holds. Using `internal` as a package risks becoming a catch-all package for anything not obviously fitting in an existing package.

### CODE REVIEW

File: internal/helpers/helpers.go [hunk 1]
```diff
@@ -4,6 +4,8 @@
 // not use this file except in compliance with the License. You may obtain
 // a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 
+// TODO: Name?
```
Comment (blink1073) [inline internal/helpers/helpers.go:7]: Why not testutil?  Are these used anywhere else?

**Review by blink1073 — APPROVED** [review]
Nice!  LGTM once the linters are satisfied

File: internal/spectest/spectest.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
```
Comment (qingyang-hu) [inline internal/spectest/spectest.go:1]:
```suggestion
// Copyright (C) MongoDB, Inc. 2023-present.
```

**Review by qingyang-hu — APPROVED** [review]
👍

**Pull request merged**